packages/jsii-dotnet-analyzers/src/Amazon.JSII.Analyzers.UnitTests/JsiiOptionalAnalyzerTests.cs

```diff
@@ -0,0 +1,114 @@
+using System.Collections.Generic;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Xunit;
+using TestHelper;
+
+namespace Amazon.JSII.Analyzers.UnitTests
+{
+    public class JsiiOptionalAnalyzerTests : DiagnosticVerifier
+    {
+        [Fact]
+        public void TestGivenSomeEmptyCodeThenRoslynDoesNotComplain()
+        {
+            var test = @"";
+
+            VerifyCSharpDiagnostic(test);
+        }
+        
+        [Fact]
+        public void TestGivenSomeCodeWithAMissingRequiredPropertyThenRoslynComplains()
+        {
+            var test = @"
+            using System;
+            namespace Amazon.JSII.Analyzers.UnitTests
+            {
+                public class JsiiByValueAttribute : Attribute
+                {
+                }
+
+                public class JsiiClassAttribute : Attribute
+                {
+                }
+
+                public class JsiiOptionalAttribute : Attribute
+                {
+                }
+
+                [JsiiByValue]
+                public class SampleProps
+                {
+                    [JsiiOptional]
+                    public string OptionalProperty1 { get; set; }
+                    
+                    [JsiiOptional]
+                    public string OptionalProperty2 { get; set; }
+                    
+                    public string RequiredProperty1 { get; set; }
+
+                    public string RequiredProperty2 { get; set; }
+                }
+
+                [JsiiClass]
+                public class SampleClass
+                {
+                    public SampleClass(SampleProps props)
+                    {
+                        props = null;
+                    }
+                }
+
+                class Test
+                {   
+                    var result1 = new SampleClass(new SampleProps());
+                    var result2 = new SampleClass(new SampleProps()
+                    {
+                        RequiredProperty1 = null,
+                        OptionalProperty2 = ""test""
+                        RequiredProperty2 = ""test""
+                    );
+                }
+            }";
+            var expected = new List<DiagnosticResult>()
+            {
+                new DiagnosticResult()
+                {
+                    Id = "CDK001",
```

Comment:
None of those should be `CDK<anything>`. Should be `JSII<anything>`

Reply:
Done!